### PR TITLE
added top-values aggregate

### PR DESCRIPTION
**Pull request by benjismith**
Added an aggregate builder for the top-values aggregate, including a few support classes and a set of tests.

### CODE REVIEW

File: src/integTest/java/io/orchestrate/client/itest/AggregateTest.java [hunk 1]
```diff
@@ -42,6 +44,106 @@
...
+        kvMetadata = client.kv(collection(), "key0").put("{`a`:1.0}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key1").put("{`a`:1.0}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key2").put("{`a`:1.0}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key3").put("{`a`:`monkey`}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key4").put("{`a`:`monkey`}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key5").put("{`a`:`monkey`}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key6").put("{`a`:`monkey`}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key7").put("{`a`:true}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key8").put("{`a`:null}".replace('`', '"')).get();
+        kvMetadata = client.kv(collection(), "key9").put("{`a`:null}".replace('`', '"')).get();
```
Comment (housejester) [inline src/integTest/java/io/orchestrate/client/itest/AggregateTest.java]: We could clean this up with a helper method. Something like addKv(String itemKey, String json_ish). Or even w/o the key for when we don't care what it is.

File: src/main/java/io/orchestrate/client/TopValuesAggregateResult.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        assert aggregateKind.equals("top_values");
+
+        int offset = json.get("offset").asInt();
+        int limit = json.get("limit").asInt();
+
+        List<CountedValue> entries = new ArrayList<CountedValue>();
+        for (JsonNode entryNode : (ArrayNode) json.get("entries")) {
+            JsonNode valueNode = entryNode.get("value");
+            long count = entryNode.get("count").asLong();
+            if (valueNode.isNull()) {
```
Comment (housejester) [inline src/main/java/io/orchestrate/client/TopValuesAggregateResult.java]: This is throwing an NPE when I test it locally. valueNode is null.

File: src/integTest/java/io/orchestrate/client/itest/AggregateTest.java [hunk 3]
```diff
@@ -42,6 +44,106 @@
...
+        Iterator<AggregateResult> i = results.getAggregates().iterator();
+        AggregateResult aggregate = i.next();
+
+        assertNotNull(aggregate);
+        assertTrue(aggregate instanceof TopValuesAggregateResult);
+        TopValuesAggregateResult topValues = (TopValuesAggregateResult) aggregate;
+        List<CountedValue> entries = topValues.getEntries();
+
+        // Check the total number of results
+        assertEquals(4L, topValues.getValueCount());
```
Comment (housejester) [inline src/integTest/java/io/orchestrate/client/itest/AggregateTest.java]: This is failing for me locally after I fix the NPE. It is getting 0L.

File: src/main/java/io/orchestrate/client/TopValuesAggregateResult.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+package io.orchestrate.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * This class represents the results of a StatsAggregate function, providing
```
Comment (housejester) [inline src/main/java/io/orchestrate/client/TopValuesAggregateResult.java]: Comment says StatsAggregate.

File: src/integTest/java/io/orchestrate/client/itest/AggregateTest.java [hunk 5]
```diff
@@ -42,6 +44,106 @@
...
```
Comment (housejester) [inline src/integTest/java/io/orchestrate/client/itest/AggregateTest.java]: This is also failing with 0 results.

Comment (housejester) [inline src/integTest/java/io/orchestrate/client/itest/AggregateTest.java]: So, it looks like there are indeed 4 entries, but the value_count is coming back as 0 from the server.

File: src/main/java/io/orchestrate/client/TopValuesAggregateResult.java (hunk 2 above)
Comment (benjismith) [inline src/main/java/io/orchestrate/client/TopValuesAggregateResult.java]: Fixed with orc PR...

File: src/integTest/java/io/orchestrate/client/itest/AggregateTest.java (hunk 3 above)
Comment (benjismith) [inline src/integTest/java/io/orchestrate/client/itest/AggregateTest.java]: Fixed with orc PR: https://github.com/orchestrate-io/orchestrate/pull/183

**Comment by housejester** [comment]
+1

**Pull request merged**